main/src/ca/uwaterloo/flix/language/phase/OccurrenceAnalyzer.scala

```diff
@@ -156,20 +363,35 @@ object OccurrenceAnalyzer {
       this.vars.getOrElse(sym, Dead)
     }
 
-    /** Returns a new [[ExpContext]] with the mapping `sym -> occur` added to [[vars]]. */
-    def addVar(sym: VarSym, occur: Occur): ExpContext = {
+    /** Returns a new [[ExprContext]] with the mapping `sym -> occur` added to [[vars]]. */
+    def addVar(sym: VarSym, occur: Occur): ExprContext = {
       this.copy(vars = this.vars + (sym -> occur))
     }
 
-    /** Returns a new [[ExpContext]] with `sym` and the corresponding value removed from [[vars]]. */
-    def removeVar(sym: VarSym): ExpContext = {
+    /** Returns a new [[ExprContext]] with `sym` and the corresponding value removed from [[vars]]. */
+    def removeVar(sym: VarSym): ExprContext = {
       this.copy(vars = this.vars - sym)
     }
 
-    /** Returns a new [[ExpContext]] with `syms` and the corresponding values removed from [[vars]]. */
-    def removeVars(syms: Iterable[VarSym]): ExpContext = {
+    /** Returns a new [[ExprContext]] with `syms` and the corresponding values removed from [[vars]]. */
+    def removeVars(syms: Iterable[VarSym]): ExprContext = {
       this.copy(vars = this.vars -- syms)
     }
+
+    /**
+      * Applies `f` to each value in `vars` where `f` is defined.
+      * Returns a new [[ExprContext]] with the updated map.
+      */
+    def map(f: PartialFunction[Occur, Occur]): ExprContext = {
```

Comment:
Just let this take a function from Occur to Occur and let the caller deal with partiality.

Reply:
Fix this too :)